chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/aggregator/workload/DurationLogProcessor.java

```diff
@@ -253,41 +237,49 @@ public StatisticsGenerator generate() throws IOException {
                     windowStatisticsCalc.addValue(logEntry.getDuration());
                     globalStatisticsCalc.addValue(logEntry.getDuration());
                 }
-            } finally {
-                if (fileReader != null) {
-                    fileReader.close();
-                }
             }
-
             if (currentCount > 0) {
                 double throughput = (double) currentCount * 1000 / intervalSize;
                 long currentTime = time - extendedInterval / 2;
-                statistics.add(new MetricPointEntity(currentTime, throughput, throughputDescription));
-                statistics.add(new MetricPointEntity(currentTime, windowStatisticsCalc.getMean() / 1000, latencyDesc));
-                statistics.add(new MetricPointEntity(currentTime, windowStatisticsCalc.getStandardDeviation() / 1000, latencyStdDevDesc));
-
-                for (Double percentileKey : getTimeWindowPercentilesKeys()) {
-                    Double value = windowStatisticsCalc.getPercentile(percentileKey) / 1000D;
-                    statistics.add(new MetricPointEntity(time, value, percentileMap.get(percentileKey)));
-                }
+                addStatistics(time, currentTime, throughput, windowStatisticsCalc, throughputDesc, latencyDesc, latencyStdDevDesc, percentiles);
```

Comment:
if there is only one time interval in the data, your hack with ++addedStatistics will not work :)